shared/src/commonMain/kotlin/org/hyperskill/app/problems_limit_info/domain/analytic/ProblemsLimitInfoModalClickedUnlockUnlimitedProblemsHSAnalyticEvent.kt

```diff
@@ -0,0 +1,34 @@
+package org.hyperskill.app.problems_limit_info.domain.analytic
+
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticAction
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticEvent
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticPart
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticRoute
+import org.hyperskill.app.analytic.domain.model.hyperskill.HyperskillAnalyticTarget
+import org.hyperskill.app.problems_limit_info.domain.analytic.ProblemsLimitInfoModalAnalyticKeys.USER_INITIATED
+import org.hyperskill.app.problems_limit_info.domain.model.ProblemsLimitInfoModalContext
+
+/**
+ * Represents click on the "Unlock unlimited problems" button in problems limit info modal analytic event.
+ *
+ * JSON payload:
+ * ```
+ * {
+ *     "route": "/learn/step/1",
+ *     "action": "click",
+ *     "part": "problems_limit_reached_modal",
+ *     "target": "unlock_unlimited_problems"
+ * }
+ * ```
+ * @see HyperskillAnalyticEvent
+ */
+class ProblemsLimitInfoModalClickedUnlockUnlimitedProblemsHSAnalyticEvent(
+    route: HyperskillAnalyticRoute,
+    context: ProblemsLimitInfoModalContext
+) : HyperskillAnalyticEvent(
+    route = route,
+    action = HyperskillAnalyticAction.CLICK,
+    part = HyperskillAnalyticPart.PROBLEMS_LIMIT_REACHED_MODAL,
+    target = HyperskillAnalyticTarget.UNLOCK_UNLIMITED_PROBLEMS,
+    context = mapOf(USER_INITIATED to (context == ProblemsLimitInfoModalContext.USER_INITIATED))
```

Comment:
missing `context` in docs, also in:
1. `ProblemsLimitInfoModalHiddenHyperskillAnalyticEvent`
2. `ProblemsLimitInfoModalShownHyperskillAnalyticEvent`